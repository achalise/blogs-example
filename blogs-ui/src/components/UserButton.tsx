import { User } from "../models";
import React from "react";

export const UserButton: React.FC<UserButtonProps> = (props) => {
    const onClick = () => {
        props.selectUser(props.user);
    }
    const classes = `btn  mr-1 ${props.selectedUser && props.selectedUser.id === props.user.id ? 'btn-warning' : 'btn-primary'}`
    return (
        <button className={classes} type="button" onClick={onClick}>{props.user.username}</button>
    );
}

export interface UserButtonProps {
    user: User;
    selectedUser?: User;
    selectUser: (user: User) => void
}