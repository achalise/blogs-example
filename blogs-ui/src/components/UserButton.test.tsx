import React from 'react';
import {render, cleanup, waitForElement} from '@testing-library/react'
import { UserButton, UserButtonProps } from './UserButton';

afterEach(cleanup);

describe(`UserButton tests`, () => {
    it(`Should correctly mark the selected user`, async (done) => {
        const testUser = {id: 1, email: "test@email.com", name: "testName", username: "testuser", posts: []}
        const props: UserButtonProps = {
            user: testUser,
            selectedUser: {...testUser, id: 2},
            selectUser: () => {}
        };

        const {container, getByText} = render(<UserButton user={props.user} selectedUser={props.user} selectUser={props.selectUser}/>);
        let element = await waitForElement(() => getByText(props.user.username));
        expect(element).toMatchSnapshot();

        const button = container.querySelector('.btn-warning');
        expect(button).toMatchSnapshot();
        done();
    })
})