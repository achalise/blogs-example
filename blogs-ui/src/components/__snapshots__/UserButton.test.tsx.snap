// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserButton tests Should correctly mark the selected user 1`] = `
<button
  class="btn  mr-1 btn-warning"
  type="button"
>
  testuser
</button>
`;

exports[`UserButton tests Should correctly mark the selected user 2`] = `
<button
  class="btn  mr-1 btn-warning"
  type="button"
>
  testuser
</button>
`;
